Content.Server/Nutrition/EntitySystems/FoodSystem.cs

```diff
@@ -227,11 +230,20 @@ private void OnDoAfter(EntityUid uid, FoodComponent component, ConsumeDoAfterEve
                 _utensilSystem.TryBreak(utensil, args.User);
             }
 
-            if (component.UsesRemaining > 0)
-            {
-                if (!forceFeed)
-                    args.Repeat = true;
+            args.Repeat = !forceFeed;
 
+            if (TryComp<StackComponent>(uid, out var stack))
+            {
+              //Not deleting whole stack piece will make troubles with grinding object
+              if (stack.Count > 1) 
+              {
+                _stack.SetCount(uid, stack.Count - 1);
+                _solutionContainerSystem.TryAddSolution(uid, solution, split);
+                return;
+              }
+            }
```

Comment:
Formatting needs fixing (4 spaces for indentation).

Also if food is injectable then might need tweaking because this clones the solution so if you inject poison into a stack of 30 for example then you get a significant amount of posion.

Reply:
it isn't injectable. It uses BaseFood insted of BaseInjectableFood.

Reply:
yes but if you make a food thats injectable and stacked it will break so you should handle it correctly and divide the solution